src/service/data_products/taxa_count.py

```diff
@@ -232,11 +255,67 @@ async def get_taxa_counts(
     q = await _query(store, collection_id, load_ver, rank)
     for dp_proc in [dp_match, dp_sel]:
         await _add_subset_data_in_place(q, store, collection_id, load_ver, rank, dp_proc)
-        # For now always sort by the std data. See if ppl want sort by match/selection data
-        # before implementing something more sophisticated.
+
+    _sort_taxa_counts(q, sort_priority, [dp_match, dp_sel])
+
     return _taxa_counts(dp_match=dp_match, dp_sel=dp_sel, data=q)
 
 
+def _fill_missing_orders(sort_order: list[str], processed_count: list[str]):
+    # fill in missing orders with the default sort order if order exists in the processed count
+    if not isinstance(sort_order, list):
+        raise ValueError(f"sort_order must be a list of strings, provided: {sort_order}")
+
+    missing_orders = [order for order in _DEFAULT_SORT_ORDER
+                      if order not in sort_order and
+                      (order in processed_count or order == names.FLD_TAXA_COUNT_COUNT)]
+
+    return missing_orders + sort_order
+
+
+def _sort_taxa_counts(
+        q: list[dict[str, Any]],
+        sort_priority: str,
+        dp_list: list[models.DataProductProcess],
+        sort_by_count_already: bool = True):
+    # Sort taxa count records in place by the sort_priority list.
+
+    processed_count = [_TYPE2FIELD[dp.type] for dp in dp_list if dp]
+
+    if sort_priority:
+        sort_priority = [p.strip() for p in sort_priority.split(",")]
+    else:
+        sort_priority = list()
+
+    if len(sort_priority) != len(set(sort_priority)):
+        raise errors.IllegalParameterError(f"Duplicate sort priority found: {sort_priority}")
+
+    try:
+        sort_order_rev = [_SORT_PRIORITY_ORDER_MAP[k] for k in sort_priority]
+    except KeyError as e:
+        raise errors.IllegalParameterError(f"Invalid sort priority: {e}, "
+                                           f"valid priorities are: {list(_SORT_PRIORITY_ORDER_MAP.keys())}")
+    missing_processes = [p for p in sort_order_rev if p not in processed_count and p != names.FLD_TAXA_COUNT_COUNT]
+    if missing_processes:
+        missing_priority = [key for key, value in _SORT_PRIORITY_ORDER_MAP.items() if value in missing_processes]
+        raise errors.IllegalParameterError(f"No occurrences of the {missing_priority} data were found in the "
+                                           f"taxa count records")
```

Comment:
This is a confusing error message - the error is that the user put match or selection in the sort priority without providing a match or selection ID

Reply:
👍

Reply:
:+1: